### ISSUE TITLE
Install scripts with package

### ISSUE THREAD

**Issue by petebachant** [py-pdf/PyPDF2#259]
This will allow their use from anywhere via command line.

**Comment by mlissner** [comment]
I only see two scripts, and I don't think they need to be installed. One is easily replicated by `pdftk`, which I expect most people will be familiar with, and the other looks pretty minimal. Are these getting a lot of use?

**Comment by switham** [comment]
Disclaimer: I wrote "pdfcat," so I'm biased, but...  

@mlissner -- which script is replicated by pdftk and which one looks minimal?  

I would think that _general usefulness_ is what ought to determine whether the scripts should be installed.  (That plus whether it's hard to set up for scripts to be installed.)

If anything, I think shortness is a plus for a script.  2-up is 55 lines and pdfcat is 81 lines.  pdfcat is based on the pagerange module (including the docstring there), so it's bigger than it looks.

Short scripts are good because the actual capabilities should be in pdftk the library, and a script is just an interface between a particular capability and the (Unix) command line.  (The fact that a script demonstrates how a Python programmer can use a feature is irrelevant to the question of whether to install the script.)

I was unfamiliar with pdftk until today.  It seems cool, but why shouldn't PyPDF2 duplicate pdftk features?  Why should someone who has PyPDF2 have to download a separate app?  And how are they supposed to know about pdftk?  If you're comparing pdftk to my pdfcat script... for one thing a script can be used within another script.  For another, some people (me included) prefer command lines to GUIs for such elemental tasks.

Zooming out, I think it would be nice if PyPDF contained a _lot_ of little utility scripts and _boasted about them on its home page_.  As a step towards this, it would be nice to set up the mechanism by which little scripts get installed, to make it easier and encourage PyPDF developers to create scripts that get installed.  2-up and pdfcat, although humble, could be the first examples of such a thing.

**Comment by mlissner** [comment]
I guess it's an identity thing. If PyPDF2 wants to have and support command line stuff, fine, but that's not usually what Python libraries are about. They're usually about providing a Python interface to doing something. I don't have a horse in this race, except that doing commandline stuff seems like a distraction for PyPDF2.

> I was unfamiliar with pdftk until today. It seems cool, but why shouldn't PyPDF2 duplicate pdftk features? 

I'm a little surprised by this, but pdftk is kind of the standard bearer for PDF manipulation. It's *very* popular and powerful. I'd say PyPDF2 shouldn't duplicate it because it's a wasted effort if there's already a powerful tool for the job. PyPDF2 could also add a script for rendering/displaying PDFs, but it shouldn't because there are lots of other apps (Acrobat, Evince, etc.) already doing that.

>  If you're comparing pdftk to my pdfcat script... for one thing a script can be used within another script. For another, some people (me included) prefer command lines to GUIs for such elemental tasks.

OK, so it seems there are two things known as pdftk, and the lame one is winning Google. That's annoying. The one I'm referring to is [this one](http://linux.die.net/man/1/pdftk). It's a command line thing quite similar to your scripts.

**Comment by lcnittl** [comment]
Speaking for a windows user: Having the pdfcat as executable in the python scrips folder (like pygmentize.exe from Pygments) would be quite amazing, as it allows the use as command line tool (as stated by the OP).

**Comment by MartinThoma** [comment]
> I guess it's an identity thing. If PyPDF2 wants to have and support command line stuff, fine, but that's not usually what Python libraries are about

I agree with that. PyPDF2 is a library, not an application.

PDF at looks cool. What do you think about creating a tiny package for it (e.g. within the py-pdf github organization)? If you need help with that, I can support :-)

**Comment by mlissner** [comment]
It's been six years since I wrote my comment above, and I think I disagree with a lot of it. I don't see any harm in including a few scripts in this package. They're minimal and useful. In fact, in the not too distant future, I wrote a Python library for doing stuff with PDFs, and, yep, I included some command line scripts.

Anyway, I disagree with myself from 2016. Scripts are handy and easy. Toss 'em in.
